README.md

```diff
@@ -1,6 +1,193 @@
 Jenkins Mercurial Plugin
 =========================
 
-This plugin integrates the Mercurial version control system with Jenkins.
+[![Jenkins Plugin](https://img.shields.io/jenkins/plugin/v/mercurial.svg)](https://plugins.jenkins.io/mercurial)
+[![GitHub release](https://img.shields.io/github/release/jenkinsci/mercurial-plugin.svg?label=release)](https://github.com/jenkinsci/mercurial-plugin/releases/latest)
+[![Jenkins Plugin Installs](https://img.shields.io/jenkins/plugin/i/mercurial.svg?color=blue)](https://plugins.jenkins.io/mercurial)
 
-See [Mercurial Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Mercurial+Plugin) on the Jenkins Wiki for more information.
+
+This plugin integrates the [Mercurial version control
+system](https://www.mercurial-scm.org/) with Jenkins.
+
+With this plugin, you can designate a Mercurial repository as the
+"upstream" repository. Every build will then run something like
+`hg pull -u` to bring the tip of this upstream repository. In a similar
+manner, polling will check if the upstream repository contains any new
+changes, and use that as the triggering condition of the new build.
+ This plugin is currently intended to support Mercurial 1.0 and later.
+
+Viewers included are `bitbucket`, `fisheye`, `google-code`, `hgweb`,
+`kiln`, and `rhodecode`.
+
+### Push Notifications
+
+As of version 1.38 it's possible to trigger builds using push
+notifications instead of polling. In your repository's .hg/hgrc file
+add:
+
+``` syntaxhighlighter-pre
```

Comment:
Can use actual Linguist tokens here BTW.

Reply:
Good point, proposed fix in #143